ci_framework/roles/polarion/tasks/main.yml

```diff
@@ -0,0 +1,111 @@
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_polarion_jump_repo_url is defined and cifmw_polarion_jump_repo_url != ""
+      - cifmw_polarion_test_id is defined and cifmw_polarion_test_id != ""
+      - cifmw_polarion_update_testcases is defined and cifmw_polarion_update_testcases | bool
+
+- name: Ensure deps are installed
+  become: true
+  tags:
+    - bootstrap
+    - packages
+  ansible.builtin.package:
+    name: "{{ cifmw_polarion_deps }}"
+    state: present
+
+- name: Look for test result XML files in artifacts directory
+  ansible.builtin.find:
+    paths: "{{ cifmw_polarion_jump_result_dir }}"
+    patterns: "*.xml"
+    recurse: yes
+  register: xml_files
+
+- name: Deploy jump and report to polarion
+  when: xml_files.files | length > 0
+  block:
+  - name: Validate XML files with xmllint
+    ansible.builtin.command: "xmllint --noout {{ item.path }}"
+    loop: "{{ xml_files.files }}"
+    register: xmllint_output
+    changed_when: false
+    failed_when: xmllint_output.rc != 0
+
+  - name: Clone Jump repository
+    ansible.builtin.git:
+      repo: "{{ cifmw_polarion_jump_repo_url }}"
+      dest: "{{ cifmw_polarion_jump_repo_dir }}"
+      version: "{{ cifmw_polarion_jump_repo_branch }}"
+      force: true
+
+  - name: Create virtualenv
+    when:
+      - cifmw_polarion_jump_repo_dir is defined and cifmw_polarion_jump_repo_dir != ""
+    ansible.builtin.pip:
+      name: pip
+      virtualenv: "{{ cifmw_polarion_jump_repo_dir }}/jump-venv"
+      virtualenv_command: "/usr/bin/python3 -m venv"
+      state: latest
+      extra_args: --upgrade
+
+
+  - name: Use polarion-staging
+    when:
+      - cifmw_polarion_fqdn is defined and cifmw_polarion_fqdn != ""
+      - cifmw_polarion_stage_fqdn is defined and cifmw_polarion_stage_fqdn != ""
+    ansible.builtin.replace:
+      path: "{{ cifmw_polarion_jump_repo_dir }}/{{ item }}"
+      regexp: "{{ cifmw_polarion_fqdn }}"
+      replace: "{{ cifmw_polarion_stage_fqdn }}"
+    loop:
+      - "helpers.py"
+      - ".pylero"
+
+  - name: Run prepare_pylero.sh
+    ansible.builtin.shell: >
+      source jump-venv/bin/activate && \
+      ./prepare_pylero.sh
+    args:
+      chdir: "{{ cifmw_polarion_jump_repo_dir }}"
+      executable: /bin/bash
+    register: pylero_output
+
+  - name: Output prepare_pylero
+    ansible.builtin.debug:
+      var: pylero_output.stdout_lines
+
+  - name: Set XML files in a fact
+    ansible.builtin.set_fact:
+      found_xml_files: "{{ xml_files.files | map(attribute='path') | join(',') }}"
```

Comment:
Do we need a fact and a task to set a fact for this? Can be a simple variable in the next task.

Reply:
thanks, fixed